lib/player.js

```diff
@@ -5750,12 +5754,104 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
+      audioTracksMap.set(track.audioId, audioTrack);
+    }
+    return Array.from(audioTracksMap.values());
+  }
+
+
+  /**
+   * Select a video track compatible with the current audio track.
+   * If the player has not loaded any content, this will be a no-op.
+   *
+   * @param {shaka.extern.VideoTrack} videoTrack
+   * @param {boolean=} clearBuffer
+   * @param {number=} safeMargin Optional amount of buffer (in seconds) to
+   *   retain when clearing the buffer. Useful for switching quickly
+   *   without causing a buffering event. Defaults to 0 if not provided. Can
+   *   cause hiccups on some browsers if chosen too small, e.g. The amount of
+   *   two segments is a fair minimum to consider as safeMargin value.
+   * @export
+   */
+  selectVideoTrack(videoTrack, clearBuffer = false, safeMargin = 0) {
+    const variants = this.getVariantTracks();
+    if (!variants.length) {
+      return;
+    }
+    const active = variants.find((t) => t.active);
+    if (!active) {
+      return;
+    }
+    const validVariant = variants.find((t) => {
+      return t.audioId === active.audioId &&
+          (t.videoBandwidth || t.bandwidth) == videoTrack.bandwidth &&
+          t.width == videoTrack.width &&
+          t.height == videoTrack.height &&
+          t.frameRate == videoTrack.frameRate &&
+          t.pixelAspectRatio == videoTrack.pixelAspectRatio &&
+          t.hdr == videoTrack.hdr &&
+          t.colorGamut == videoTrack.colorGamut &&
+          t.videoLayout == videoTrack.videoLayout &&
+          t.videoMimeType == videoTrack.mimeType &&
+          t.videoCodec == videoTrack.codecs;
+    });
+    if (validVariant && !validVariant.active) {
+      this.selectVariantTrack(validVariant, clearBuffer, safeMargin);
     }
-    if (!audioTracksSet.size) {
+  }
+
+
+  /**
+   * Return a list of video tracks compatible with the current audio track.
+   *
+   * @return {!Array<shaka.extern.VideoTrack>}
+   * @export
+   */
+  getVideoTracks() {
+    if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
       return [];
     }
-    return Array.from(audioTracksSet);
+    const variants = this.getVariantTracks();
+    if (!variants.length) {
+      return [];
+    }
+    const active = variants.find((t) => t.active);
+    if (!active) {
+      return [];
+    }
+    const filteredTracks = variants.filter((t) => {
+      return t.originalAudioId === active.originalAudioId &&
+          t.audioId === active.audioId &&
+          t.audioGroupId === active.audioGroupId &&
+          t.videoCodec;
+    });
+    if (!filteredTracks.length) {
+      return [];
+    }
+
+    /** @type {!Map<number, shaka.extern.VideoTrack>} */
+    const videoTracksMap = new Map();
+    for (const track of filteredTracks) {
+      if (track.videoId == null) {
+        continue;
+      }
+      /** @type {shaka.extern.VideoTrack} */
+      const videoTrack = {
+        active: track.active,
+        bandwidth: track.videoBandwidth || track.bandwidth,
```

Comment:
track.bandwidth would be actually audio bandiwdth in this case, isn't it? Shouldn't we set it to null / 0 / NaN?

Reply:
In the UI we used this, because in HLS there is no way to differentiate the bandwidth of audio and video, so for compatibility, we have to keep this here, I know it is not the best, but it is the best way to adapt to HLS and DASH.